bcm63xx: Add support for D-Link DSL-2750u rev C1

naseef opened the pull request:
This adds support for the D-Link DSL-2750u rev C1.
(https://deviwiki.com/wiki/D-Link_DSL-2750U_rev_C1)

It uses the same hardware as ADB P.DG A4001N.
CPU:   Broadcom BCM63281 (320 MHz)
RAM:   32M (Winbond W9725G6JB)
Flash:   8M (MXIC MX25L6445E)
Ethernet:   4x 100 Mbps
Wireless:   802.11b/g/n: BCM43225
USB:   1x 2.0

Flash instructions:

1.  Assign static IP 192.168.1.100 to PC
2.  Unplug the power source
3.  Press the RESET button at the router, don't release it yet!
4.  Plug the power source.Wait some seconds
5.  Release the RESET button
6.  Browse to http://192.168.1.1
7.  Send the openwrt-bcm63xx-generic-DSL2750U-C1-squashfs-cfe.bin and
    wait some minutes until the firmware upgrade finish.

adschm commented on target/linux/bcm63xx/patches-5.4/569-board_dsl-2750u-c1.patch:
```diff
@@ -0,0 +1,68 @@
+ 	},
+ };
+ 
++static struct board_info __initdata board_dsl_2750u_c1 = {
```
Do you really need a full new definition here?

naseef replied:
Sorry. I'm new to this stuff..can you please elaborate?

adschm replied:
You say it uses the same hardware as the a4001n. So, why just copy/paste the same definitions again? You could just add the compatible for this device in 528-board_bcm6328-adb-a4001n.patch and use the existing board definition there:

+	{ .compatible = "d-link,dsl-2750u-c1", .data = &board_A4001N, },

adschm commented on target/linux/bcm63xx/base-files/etc/board.d/01_leds:
```diff
@@ -43,6 +43,9 @@ comtrend,vr-3032u|\
 	;;
+d-link,dsl-2750u-c1)
+	ucidef_set_led_usbdev "usb" "USB" "dsl-2750u-c1:green:usb" "1-1"
+	;;
```
Just add your device to the block directly above, as $model will evaluate to dsl-2750u-c1.

adschm commented:
You need to add a Signed-off-by to your commit message. Despite, please extend the Specifications list.

naseef committed:
bcm63xx: Add support for D-Link DSL-2750u rev C1

This adds support for the D-Link DSL-2750u rev C1.
(https://deviwiki.com/wiki/D-Link_DSL-2750U_rev_C1)

It uses the same hardware as ADB P.DG A4001N.
CPU:   Broadcom BCM63281 (320 MHz)
RAM:   32M (Winbond W9725G6JB)
Flash:   8M (MXIC MX25L6445E)
Ethernet:   4x 100 Mbps
Wireless:   802.11b/g/n: BCM43225
USB:   1x 2.0

Flash instructions:

1.  Assign static IP 192.168.1.100 to PC
2.  Unplug the power source
3.  Press the RESET button at the router, don't release it yet!
4.  Plug the power source.Wait some seconds
5.  Release the RESET button
6.  Browse to http://192.168.1.1
7.  Send the openwrt-bcm63xx-generic-DSL2750U-C1-squashfs-cfe.bin and
    wait some minutes until the firmware upgrade finish.

Signed-off-by: Ahmed Naseef <naseefkm@gmail.com>

naseef commented:
Incorporated suggeste changes

adschm commented on target/linux/bcm63xx/dts/bcm6328-d-link-dsl-2750u-c1.dts:
```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
```
dts-v1 can be dropped.

adschm commented on target/linux/bcm63xx/dts/bcm6328-d-link-dsl-2750u-c1.dts:
```diff
@@ -0,0 +1,128 @@
+		compatible = "gpio-leds";
+
+		inet_green {
+			label = "dsl-2750u-c1:green:inet";
```
model prefix can be dropped from LED names, leaving only "green:inet" etc.

Noltari commented:
Merged with some modifications, thanks!
https://github.com/openwrt/openwrt/commit/8eeb57acb75f9d89abed59595de052dba9fed7b8
